test/Sanitizers/tsan-norace-deinit-run-time.swift

```diff
@@ -0,0 +1,47 @@
+// RUN: %target-swiftc_driver %s -g -sanitize=thread -o %t_tsan-binary
+// RUN: env TSAN_OPTIONS=abort_on_error=0:ignore_interceptors_accesses=1 %target-run %t_tsan-binary 2>&1 | %FileCheck %s
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+// REQUIRES: CPU=x86_64
+// REQUIRES: tsan_runtime
+// XFAIL: linux
+
+// Test that we do not report a race on deinit; the syncronization is guaranteed by runtime.
+import Foundation
+
+public class TestDeallocObject : NSObject {
+  public var v : Int
+  public override init() {
+    v = 1
+  }
+
+  func accessMember() {
+    var local : Int = v
```

Comment:
This will get optimized out when you run check-swift-optimize -- pass the result of the load to another function and mark it as @_semantics("optimize.sil.never")